great_tables/_scss.py

```diff
@@ -39,23 +39,30 @@
 )
 
 
-def font_color(color: str, table_font_color: str, table_font_color_light: str):
-    if color.startswith("#"):
-        rgb = wc.hex_to_rgb(color)
-    elif color.startswith("rgb") and "%" in color:
-        # TODO: rgb_percent_to_rgb() expects a tuple
-        raise NotImplementedError()
-        rgb = wc.rgb_percent_to_rgb(color)
-    else:
-        rgb = wc.name_to_rgb(color)
+def _font_color(color: str, table_font_color: str, table_font_color_light: str) -> str:
 
-    if (rgb.red * 0.299 + rgb.green * 0.587 + rgb.blue * 0.114) > 186:
+    if color == "transparent":
+        # With the `transparent` color, the font color should have the same value
+        # as the `table_font_color` option since the background will be transparent
         return table_font_color
+    if color in ["currentcolor", "currentColor", "inherit", "initial", "unset"]:
+        # For the other valid CSS color attribute values, we should pass them through
+        return color
+
+    # Normalize the color to a hexadecimal value
+    color_normalized = _html_color(colors=[color])
+
+    # Determine the ideal font color given the different background colors
+    ideal_font_color = _ideal_fgnd_color(
+        bgnd_color=color_normalized[0],
+        light=table_font_color,
+        dark=table_font_color_light,
+    )
 
-    return table_font_color_light
+    return ideal_font_color
 
 
-def css_add(value: str | int, amount: int):
```

Comment:
I'm still not entirely clear on whether to consistently use leading underscores for 'private' functions/methods. Maybe we could talk about this briefly so I could better understand whether it's needed virtually everywhere.

Reply:
Let's discuss. These types of changes aren't really related to the PR, so I would push them out of scope (and then confirm in words the refactor changes when the PR is reviewed. That way the relevant fix can go through review quickly, and refactor changes can be applied at the end).

Making these kinds of changes brings a lot of extra files into the PR review